api/src/pcapi/domain/password.py

```diff
@@ -54,21 +71,24 @@ def _ensure_new_password_is_strong_enough(field_name: str, field_value: str, err
     regex = "^" + at_least_one_uppercase + at_least_one_lowercase + at_least_one_digit + min_length + "$"
 
     # Special characters: !"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~
-    at_least_one_special_char = len(set(field_value).intersection(set(string.punctuation))) > 0
-
-    if not re.match(regex, field_value) or not at_least_one_special_char:
-        errors.add_error(
-            field_name,
-            "Le mot de passe doit contenir au moins :\n"
-            "- 12 caractères\n"
-            "- Un chiffre\n"
-            "- Une majuscule et une minuscule\n"
-            "- Un caractère spécial",
-        )
-
-
-def _ensure_confirmation_password_is_same_as_new_password(
-    new_password_value: str, new_confirmation_password_value: str, errors: ApiErrors
-) -> None:
+    at_least_one_special_char = len(set(password).intersection(set(string.punctuation))) > 0
+
+    if not re.match(regex, password) or not at_least_one_special_char:
+        return {
+            field_name: [
+                "Le mot de passe doit contenir au moins :\n"
+                "- 12 caractères\n"
+                "- Un chiffre\n"
+                "- Une majuscule et une minuscule\n"
+                "- Un caractère spécial"
+            ]
+        }
+    return {}
```

Comment:
Suite à mes remarques précédentes, ça donnerait : 
```suggestion
        return [
                "Le mot de passe doit contenir au moins :\n"
                "- 12 caractères\n"
                "- Un chiffre\n"
                "- Une majuscule et une minuscule\n"
                "- Un caractère spécial"
            ]
    return []
```